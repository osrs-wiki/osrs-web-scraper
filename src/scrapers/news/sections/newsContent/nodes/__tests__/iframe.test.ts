import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import iframeParser from "../iframe";

describe("iframe node", () => {
  test("iframe with youtube link should use Youtube template", () => {
    const root = parse(
      '<iframe data-cookieblock-src="https://www.youtube.com/embed/80pYF2u-lJo?si=eQouUlth2Sko8b46">blah</iframe>'
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([iframeParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("iframe with youtube playlist link should use Youtube template with type=playlist", () => {
    const root = parse(
      '<iframe data-cookieblock-src="https://www.youtube.com/embed/videoseries?si=rh-lb4jty9OTAKAM&list=PLRs68iqW7gYvSyTqu12WFMMXYZM-2regV">blah</iframe>'
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([iframeParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("iframe with non-youtube link should use center tag", () => {
    const root = parse(
      '<iframe data-cookieblock-src="https://example.com">blah</iframe>'
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([iframeParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("iframe link should be parsed from src attribute", () => {
    const root = parse('<iframe src="https://example.com">blah</iframe>');
    const builder = new MediaWikiBuilder();
    builder.addContents([iframeParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
