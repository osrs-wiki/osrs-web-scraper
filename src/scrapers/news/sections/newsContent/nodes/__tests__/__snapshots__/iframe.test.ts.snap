// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iframe node iframe with non-youtube link should use center tag 1`] = `"<center><big>[https://example.com title]</big></center>"`;

exports[`iframe node iframe with youtube link should use Youtube template 1`] = `
"{{Youtube|test}}
"
`;
